networkx/algorithms/centrality/laplacian.py

```diff
@@ -114,6 +114,11 @@ def laplacian_centrality(
 
     full_energy = np.power(sp.linalg.eigh(lap_matrix, eigvals_only=True), 2).sum()
 
+    if full_energy == 0.0 and normalized:
+        raise nx.NetworkXPointlessConcept(
+            "graph without edges has centrality undefined or 0 and can not be be normalized"
+        )
+
```

Comment:
IMO it'd be better to avoid a floating-point comparison with 0.0 if possible; especially since the linalg routines very regularly give values a few ULP away from 0.

IIUC, it seems like checking the number of edges in the subgraph defined by `nodes` is equivalent? I.e. something like `if nx.subgraph(G, nbunch=nodes).number_of_edges() == 0`. If that's the case, then I'd prefer that check to avoid floating point issues. Since it doesn't rely on the Laplacian, the check can also be performed before the Laplacian is computed.